Eatery Blue/Eatery Blue/UI/HomeScreen/HomeModelController.swift

```diff
@@ -264,7 +264,7 @@ class HomeModelController: HomeViewController {
         Task {
             await withTaskGroup(of: Void.self) { [self] group in
                 group.addTask {
-                    await updateAllEateriesFromNetworking()
+                    await self.updateAllEateriesFromNetworking()
                 }
```

Comment:
You are using a strong reference to `self` within the closure which can cause retain cycles. Change `[self] group in` to `[weak self] group in` to create a weak reference of this object when it gets captured. You will then need to change line 267 to `await self?.updateAllEateriesFromNetworking()`